### PR TITLE
feat: add encrypted value in the transaction output

### PR DESCRIPTION
**Pull request by therustmonk**
Description
---
The PR adds extra encrypted field to `UnblindedOutput` structs.

What was done:
- Added `EncryptedValue` struct
- Implemented `ConsensusEncode`, `ConsensusDecode` and `ByteArray` for it
- Added the `encrypted_value` field of that type to the `UnblindedOutput` struct
- Added an extra parameter to metadata signature calculation methods
- Added an extra field to the `GRPC` protocol (`UnblindedOutput` type)
- Added an extra column and migrations for the `outputs` table
- The types `OutputSql` and `NewOutputSql` updated to store and restore the `EncryptedValue`

Important notes:
1. The value of `encrypted_value` passed as a parameter to the metadata signature evaluation but has not taken into account yet, because it will break the verification process of the `TransactionOutput` and we can complete that when the same `encrypted_value` field will be added to the `TransactionOutput` struct in the further PRs.
2. Real encryption process is not implemented yet. Instead of that I've added the method `todo_encrypt_from` that I used to mark the places where we have to put an encryption service and call it.

Motivation and Context
---
Add an encrypted amount to the `TransactionOutput`.

How Has This Been Tested?
---
CI

### CODE REVIEW

**Comment by hansieodendaal** [comment]
I see `cargo clippy --all-targets -- -D warnings` picks up some errors, probably from an earlier merge, not too many, could you please fix it?

**Review by hansieodendaal — DISMISSED** [review]
LGTM!

File: base_layer/core/src/transactions/transaction_components/transaction_output.rs [hunk 1]
```diff
@@ -261,6 +266,7 @@ impl TransactionOutput {
                 .chain(sender_offset_public_key)
                 .chain(commitment)
                 .chain(covenant)
+                //.chain(encrypted_value)
```
Comment (stringhandler) [inline base_layer/core/src/transactions/transaction_components/transaction_output.rs:269]: Should this not be in here?

Comment (therustmonk) [reply base_layer/core/src/transactions/transaction_components/transaction_output.rs:269]: It is, but to add that I have to add `encrypted_value` to the TransactionOutput as well. I'm going to do that in the next PR to keep this one smaller.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**